diff --git a/render/fx_renderer/fx_texture.c b/render/fx_renderer/fx_texture.c
index 5f447ac..153e8f4 100644
--- a/render/fx_renderer/fx_texture.c
+++ b/render/fx_renderer/fx_texture.c
@@ -248,6 +248,10 @@ static uint32_t fx_texture_preferred_read_format(struct wlr_texture *wlr_texture)
 		get_fx_format_from_gl(gl_format, gl_type, alpha_size > 0);
 	if (pix_fmt != NULL) {
 		fmt = pix_fmt->drm_format;
+		if (fmt == DRM_FORMAT_BGR888 &&
+				texture->fx_renderer->exts.EXT_read_format_bgra) {
+			fmt = DRM_FORMAT_XRGB8888;
+		}
 		goto out;
 	}
 
